ex1/dates_times.py

```diff
@@ -29,16 +29,34 @@
 '''
 Exercises for using the datetime and the calendar module
 '''
+from datetime import date, datetime, timedelta, time
 
 # Define a function named last_of_month that takes an argument dt of type date
 # and returns a date object representing the last day of the month dt was in.
+def last_of_month( dt ):
+    return ( date( dt.year, dt.month+1, 1 ) - date( dt.year, dt.month, 1 ))
+
+# a = date(2012, 2, 18)
+# print( type( last_of_month( a )))
 
 # Define a function named feed_the_gremlin which takes a time object as an
 # argument. It should return False between midnight and 6:30AM and True
 # otherwise.
+def feed_the_gremlin( dt ):
+    return dt > time(0, 0, 0) and dt < time( 6, 30, 0 )
+
+#if feed_the_gremlin( time( 6, 30, 0 )):
+#    print( "yess" )
 
 # Define a function named how_long that takes two datetime objects dt and ref
 # where ref is the reference datetime, calculates the difference between them and
 # returns the difference as a string formatted like:
 # "01 days, 01 minutes, 01 seconds until 2000-12-31 15:59:59"
 # If ref is before dt then use 'since' instead of 'until'
+def how_long( dt, ref ):
+    strmid = "until"
+    delta = dt - ref
+    if ( ref < dt ):
+        strmid = "since"
+    return delta.days + " days, " + delta.minutes + " minutes, " + delta.seconds + " seconds " + strmid + ref.strftime("%Y-%m-%d %H:%M:%S")
+print( how_long( date.today, date( 1016, 3, 5 )))
```

Comment:
This only works with datetime objects thats one reason why the tests for this module fail. 

You should not have a print statement on this level. If you need it then put it after `if __name__ == "__main__"`